metrics/prometheus.go

```diff
@@ -0,0 +1,348 @@
+			[]string{"name", "id"},
+			nil),
+		cpuUsageSeconds: prometheus.NewDesc(
+			"container_cpu_usage_seconds_total",
+			"Total seconds of cpu time consumed.",
+			[]string{"name", "id", "type"},
+			nil),
+		cpuUsageSecondsPerCPU: prometheus.NewDesc(
+			"container_cpu_usage_seconds_total",
+			"Total seconds of cpu time consumed per cpu.",
+			[]string{"name", "id", "cpu"},
+			nil),
+
+		memoryUsageBytes: prometheus.NewDesc(
+			"container_memory_usage_bytes",
+			"Current memory usage in bytes.",
+			[]string{"name", "id"},
+			nil),
+		memoryWorkingSet: prometheus.NewDesc(
+			"container_memory_working_set_bytes",
+			"Current working set in bytes.",
+			[]string{"name", "id"},
+			nil),
+		memoryFailures: prometheus.NewDesc(
+			"container_memory_failures_total",
+			"Number of memory allocation failures.",
+			[]string{"type", "scope", "name", "id"},
+			nil),
+
+		fsLimit: prometheus.NewDesc(
+			"container_fs_limit_bytes",
+			"Number of bytes that can be consumed by the container on this filesystem.",
+			[]string{"name", "id", "device"},
+			nil),
+		fsUsage: prometheus.NewDesc(
+			"container_fs_usage_bytes",
+			"Number of bytes that are consumed by the container on this filesystem.",
+			[]string{"name", "id", "device"},
+			nil),
+		fsReads: prometheus.NewDesc(
+			"container_fs_reads_total",
+			"Number of reads completed",
+			[]string{"name", "id", "device"},
+			nil),
+		fsReadsSectors: prometheus.NewDesc(
+			"container_fs_sector_reads_total",
+			"Number of sector reads completed",
+			[]string{"name", "id", "device"},
+			nil),
+		fsReadsMerged: prometheus.NewDesc(
+			"container_fs_reads_merged_total",
+			"Number of reads merged",
+			[]string{"name", "id", "device"},
+			nil),
+		fsReadTime: prometheus.NewDesc(
+			"container_fs_read_seconds",
+			"Number of seconds spent reading",
+			[]string{"name", "id", "device"},
+			nil),
+		fsWrites: prometheus.NewDesc(
+			"container_fs_writes_total",
+			"Number of writes completed",
+			[]string{"name", "id", "device"},
+			nil),
+		fsWritesSectors: prometheus.NewDesc(
+			"container_fs_sector_writes_total",
+			"Number of sector writes completed",
+			[]string{"name", "id", "device"},
+			nil),
+		fsWritesMerged: prometheus.NewDesc(
+			"container_fs_writes_merged_total",
+			"Number of writes merged",
+			[]string{"name", "id", "device"},
+			nil),
+		fsWriteTime: prometheus.NewDesc(
+			"container_fs_write_seconds",
+			"Number of seconds spent writing",
+			[]string{"name", "id", "device"},
+			nil),
+		fsIoInProgress: prometheus.NewDesc(
+			"container_fs_io_current",
+			"Number of I/Os currently in progress",
+			[]string{"name", "id", "device"},
+			nil),
+		fsIoTime: prometheus.NewDesc(
+			"container_fs_io_time_seconds_total",
+			"Number of seconds spent doing I/Os",
+			[]string{"name", "id", "device"},
+			nil),
+		fsWeightedIoTime: prometheus.NewDesc(
+			"container_fs_io_time_weighted_seconds_total",
+			"Number of seconds spent doing I/Os",
```

Comment:
Should this and some of the counters above also start with "Cumulative count of...", like the counters below? That makes it clear to the user that they are indeed cumulative and not over some arbitrary period of time.